store/postgres/src/entities.rs

```diff
@@ -823,6 +832,28 @@ fn find_schema(
         .optional()?)
 }
 
+fn supports_proof_of_indexing(
+    conn: &diesel::pg::PgConnection,
+    subgraph_id: &SubgraphDeploymentId,
+    schema: &str,
+) -> Result<bool, StoreError> {
+    if subgraph_id == &*SUBGRAPHS_ID {
+        return Ok(false);
+    }
+    #[derive(Debug, QueryableByName)]
+    struct Table {
+        #[sql_type = "Text"]
+        pub table_name: String,
+    }
+    let query =
+        "SELECT table_name FROM information_schema.tables WHERE table_schema=$1 AND table_name=$2";
+    let result: Vec<Table> = diesel::sql_query(query)
+        .bind::<Text, _>(schema)
+        .bind::<Text, _>("poi_artifact")
+        .load(conn)?;
+    Ok(result.len() > 0)
+}
+
```

Comment:
I wonder if that should go into the `metadata` module that @lutter is preparing in the grafting PR?

Reply:
Yeah, that would be a good place for it (this query could also be written using Diesel directly, but I don't want @That3Percent have to go through the whole `table!` etc. dance I can fix that up once this is committed. We should just remember to do that.